skyportal/handlers/api/spatial_catalog.py

```diff
@@ -0,0 +1,379 @@
+          200:
+            content:
+              application/json:
+                schema:
+                  allOf:
+                    - $ref: '#/components/schemas/Success'
+                    - type: object
+                      properties:
+                        data:
+                          type: object
+                          properties:
+                            id:
+                              type: integer
+                              description: New spatial catalog ID
+          400:
+            content:
+              application/json:
+                schema: Error
+        """
+
+        data = self.get_json()
+        catalog_name = data.get('catalog_name')
+        catalog_data = data.get('catalog_data')
+
+        if catalog_name is None:
+            return self.error("catalog_name is a required parameter.")
+        if catalog_data is None:
+            return self.error("catalog_data is a required parameter.")
+
+        # check RA bounds
+        if any([(x < 0) or (x >= 360) for x in catalog_data['ra']]):
+            return self.error("ra should span 0=<ra<360.")
+
+        # check Declination bounds
+        if any([(x > 90) or (x < -90) for x in catalog_data['dec']]):
+            return self.error("declination should span -90<dec<90.")
+
+        # check for cone or ellipse keys
+        if (not {'radius'}.issubset(set(catalog_data.keys()))) and (
+            not {'amaj', 'amin', 'phi'}.issubset(set(catalog_data.keys()))
+        ):
+            return self.error("error or amaj, amin, and phi required in field_data.")
+
+        with self.Session() as session:
+            stmt = SpatialCatalog.select(self.current_user).where(
+                SpatialCatalog.catalog_name == catalog_name
+            )
+            catalog = session.scalars(stmt).first()
+            if catalog is None:
+                catalog = SpatialCatalog(catalog_name=catalog_name)
+                session.add(catalog)
+                session.commit()
+
+            IOLoop.current().run_in_executor(
+                None, lambda: add_catalog(catalog.id, catalog_data)
+            )
+
+            return self.success(data={"id": catalog.id})
+
+    @auth_or_token
+    async def get(self, catalog_id=None):
+        """
+        ---
+        single:
+          description: Retrieve a SpatialCatalog
+          tags:
+            - spatial_catalogs
+          parameters:
+            - in: path
+              name: catalog_id
+              required: true
+              schema:
+                type: integer
+        multiple:
+          description: Retrieve all SpatialCatalogs
+          tags:
+            - spatial_catalogs
+        """
+
+        catalog_name = self.get_query_argument("catalog_name", None)
+
+        with self.Session() as session:
+
+            if catalog_id is not None:
+
+                stmt = SpatialCatalog.select(self.current_user).where(
+                    SpatialCatalog.id == int(catalog_id)
```

Comment:
instead of converting to int in the query, I would do that right after getting the `catalog_name` at the beginning of the handler. In general you could verify the datatype of your query params and data before running the queries.

Reply:
good idea.